bundles/http/Citrix/Microsoft-Outlook/script.js

```diff
@@ -1,321 +1,832 @@
-		}
-		const myEventResponse = await myEventRequest.json();
-		if (myEventRequest.ok) {
-			for (const myEventsValue of myEventResponse.value) {
-				dataStore.save('my_events', {
-					i_cal_u_id: myEventsValue?.iCalUId ?? null,
-					id: myEventsValue?.id ?? null,
-					recurrence_pattern_day_of: myEventsValue?.recurrence?.pattern?.dayOfMonth ?? null,
-					recurrence_pattern_type: myEventsValue?.recurrence?.pattern?.type ?? null,
-					recurrence_range_end_date: myEventsValue?.recurrence?.range?.endDate ?? null
-				});
-			}
-		}
-	}
+  for (const id of userId) {
+    const myEventRequest = await client.fetch(
+      `/v1.0/users/${id}/calendar/events?$top=${pageSize}`
+    );
+    if (!myEventRequest.ok && myEventRequest.status != 404) {
+      throw new Error(
+        `Events sync failed ${myEventRequest.status}:${myEventRequest.statusText}`
+      );
+    }
+    const myEventResponse = await myEventRequest.json();
+    if (myEventRequest.ok) {
+      for (const myEventsValue of myEventResponse.value) {
+        dataStore.save("my_events", {
+          i_cal_u_id: myEventsValue?.iCalUId ?? null,
+          id: myEventsValue?.id ?? null,
+          recurrence_pattern_day_of:
+            myEventsValue?.recurrence?.pattern?.dayOfMonth ?? null,
+          recurrence_pattern_type:
+            myEventsValue?.recurrence?.pattern?.type ?? null,
+          recurrence_range_end_date:
+            myEventsValue?.recurrence?.range?.endDate ?? null,
+        });
+      }
+    }
+  }
+}
+//CreateRecurringOfficeHourswithCustomTimezone
+async function createRecurringOfficeHourswithCustomTimezone({
+  dataStore,
+  client,
+  actionParameters,
+}) {
+  const startDate = moment.utc().subtract(1, "m").format();
+
+  const responseOfSwitchCoustomTimeZone = await client.fetch(`v1.0/me/events`, {
+    method: "POST",
+    body: JSON.stringify({
+      subject: actionParameters.subject,
+      body: {
+        contentType: "HTML",
+        content: actionParameters.content,
+      },
+      start: {
+        dateTime: `${actionParameters.startDate}T${actionParameters.startTime}`,
+        timeZone: actionParameters.timezone,
+      },
+      end: {
+        dateTime: `${actionParameters.endDate}T${actionParameters.endTime}`,
+        timeZone: actionParameters.timezone,
+      },
+      allowNewTimeProposals: true,
+      isOnlineMeeting: actionParameters.isOnlineMeeting,
+      onlineMeetingProvider: "teamsForBusiness",
+      recurrence: {
+        pattern: {
+          type: actionParameters.recurrencetype,
+          interval: 1,
+          daysOfWeek: [actionParameters.days],
+          dayOfMonth: actionParameters.dayOfMonth,
+        },
+        range: {
+          type: "endDate",
+          startDate: moment(actionParameters.startDate).format("YYYY-MM-DD"),
+          endDate: actionParameters.recurEndDate,
+        },
+      },
+    }),
+  });
+  if (!responseOfSwitchCoustomTimeZone.ok) {
```

Comment:
Do spell check the code.
```suggestion
  if (!responseOfSwitchCustomTimeZone.ok) {
```